core/registration.c

```diff
@@ -526,7 +572,11 @@ lwm2m_status_t registration_getStatus(lwm2m_context_t * contextP)
             case STATE_REG_PENDING:
                 reg_status = STATE_REG_PENDING;
                 break;
-
+            case STATE_REG_PARTIAL_PROGRESS:
```

Comment:
Seems to me this should just return STATE_REG_PENDING.

Reply:
I have added two registration states viz.
1. STATE_REG_PARTIAL_PROGRESS: This indicates that a block of data has been sent, but 2.31 Continue response is yet to be received from the server

2. STATE_REG_PARTIAL_DONE: This would indicate that 2.31 response has been received for that block.

During block transfer registration will oscillate between these states for multiple blocks till the last block is sent where the state changes to STATE_REG_PENDING

Reply:
I understand that, I just think it shouldn't be exposed to the calling functions. I think it can be handled entirely within the registration processing. Either the reply can trigger sending the next block or it could be done in registration_step.

If it stays this way, the comment just above the start of registration_getStatus needs to be updated to indicate the new return values.

Reply:
Thanks, and agree, it can be done without exposing it to liblwm2m. Will check this out too.